enhancements/installer/ipv6-single-stack-and-dual-stack-enablement.md

```diff
@@ -0,0 +1,437 @@
+the subnets that you wish to create. When these variables are not present the GCP provider will use default values.
+
+```go
+subnet := &compute.Subnetwork{
+	Name:      subnetwork.Name,
+	Region:    subnetwork.Region,
+        ...
+	StackType: subnetwork.StackType,      << Added feature
+}
+
+if subnetwork.StackType == "IPV4_ONLY" || subnetwork.StackType == "IPV4_IPV6" {
+    secondaryIPRanges := []*compute.SubnetworkSecondaryRange{}
+    for rangeName, secondaryCidrBlock := range subnetwork.SecondaryCidrBlocks {
+        secondaryIPRanges = append(secondaryIPRanges, &compute.SubnetworkSecondaryRange{RangeName: rangeName, IpCidrRange: secondaryCidrBlock})
+    }
+    subnet.SecondaryIpRanges = secondaryIPRanges
+    subnet.IpCidrRange = subnetwork.CidrBlock
+}
+
+if subnetwork.StackType == "IPV6_ONLY" || subnetwork.StackType == "IPV4_IPV6" {
+    subnet.ExternalIpv6Prefix = subnetwork.ExternalIpv6Prefix
+    subnet.InternalIpv6Prefix = subnetwork.InternalIpv6Prefix
+    subnet.Ipv6AccessType = subnetwork.Ipv6AccessType
+}
+```
+
+##### Firewalls
+
+The firewall rules should be adjusted to follow the IPv4 and IPv6 addresses and ports found [here](https://cloud.google.com/load-balancing/docs/health-check-concepts#ip-ranges).
+
+The IPv4 Addresses are the following:
+
+```go
+SourceRanges: []string{"35.191.0.0/16", "130.211.0.0/22"},
+```
+
+The IPv6 Addresses are the following:
+
+```go
+SourceRanges: []string{"2600:2d00:1:b029::/64", "2600:2d00:1:1::/64"}
+```
+
+_Note_: A firewall rule cannot have IPv4 and IPv6 Addresses, so multiple rules must be created for dual stack instances.
+
+##### Machines
+
+The Network and AccessConfig values for Machines must be edited based on the stack type.
+
+```go
+name := "External NAT"
+configType := "ONE_TO_ONE_NAT"
+if m.GCPMachine.Spec.StackType == "IPV6_ONLY" {
+    name = "External IPv6"
+    configType = "DIRECT_IPV6"
+}
+
+networkInterface.AccessConfigs = []*compute.AccessConfig{
+{
+    Type: configType,
+    Name: name,
+},
+```
+
+##### Load Balancers
+
+TODO
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Dual stacks are currently supported by Hypershift.
+
+#### Standalone Clusters
+
+This features should cause no resource limitations for Standalone Clusters.
+
+#### Single-node Deployments or MicroShift
+
+This feature should cause no resources limitations in Single Node or MicroShift clusters.
+
+### Implementation Details/Notes/Constraints
+
+#### Install Config
+
+##### AWS
+
+TODO
+
+##### Azure
+
+TODO
+
+##### GCP
+
+The platform section, for GCP, will all contain a field for `stackType`. The `clusterNetwork` and 
+`machineNetwork` values in the `networking` section will accept IPv6 Prefix strings.
+
+```yaml
+networking:
+  clusterNetwork:
+    - cidr: 10.128.0.0/14  <<< Accept IPv6 Prefix
+      hostPrefix: 23
+  machineNetwork:
+    - cidr: 10.0.0.0/16  <<< Accept IPv6 Prefix
```

Comment:
Could this be updated to include some sample IPv6 prefixes? I'm curious to see what a customer might actually put here.

It looks like it would be something like the (routing prefix + subnet ID) prefix mentioned here? https://en.wikipedia.org/wiki/IPv6_address#Address_space